src/server/pps/cmds/cmds.go

```diff
@@ -538,6 +539,84 @@ All jobs created by a pipeline will create commits in the pipeline's repo.
 	}
 	rawFlag(inspectPipeline)
 
+	extractPipeline := &cobra.Command{
+		Use:   "extract-pipeline pipeline-name",
+		Short: "Return the manifest used to create a pipeline.",
+		Long:  "Return the manifest used to create a pipeline.",
+		Run: cmdutil.RunFixedArgs(1, func(args []string) error {
+			client, err := pachdclient.NewOnUserMachine(metrics, "user")
+			if err != nil {
+				return err
+			}
+			createPipelineRequest, err := client.ExtractPipeline(args[0])
+			if err != nil {
+				return err
+			}
+			return marshaller.Marshal(os.Stdout, createPipelineRequest)
+		}),
+	}
+
+	var editor string
+	editPipeline := &cobra.Command{
+		Use:   "edit-pipeline pipeline-name",
+		Short: "Edit the manifest for a pipeline in your text editor.",
+		Long:  "Edit the manifest for a pipeline in your text editor.",
+		Run: cmdutil.RunFixedArgs(1, func(args []string) (retErr error) {
+			client, err := pachdclient.NewOnUserMachine(metrics, "user")
+			if err != nil {
+				return err
+			}
+			createPipelineRequest, err := client.ExtractPipeline(args[0])
+			if err != nil {
+				return err
+			}
+			f, err := ioutil.TempFile("", args[0])
+			if err != nil {
+				return err
+			}
+			if err := marshaller.Marshal(f, createPipelineRequest); err != nil {
+				return err
+			}
+			defer func() {
+				if err := f.Close(); err != nil && retErr == nil {
+					retErr = err
+				}
+			}()
+			if editor == "" {
+				editor = os.Getenv("EDITOR")
+			}
+			if editor == "" {
+				editor = "vim"
+			}
+			if err := cmdutil.RunIO(cmdutil.IO{
+				Stdin:  os.Stdin,
+				Stdout: os.Stdout,
+				Stderr: os.Stderr,
+			}, editor, f.Name()); err != nil {
+				return err
+			}
+			cfgReader, err := ppsutil.NewPipelineManifestReader(f.Name())
```

Comment:
So when I do `kubectl edit ...` and I don't make any changes, nothing gets updated. It seems in this code, even if you make no changes, we'll update (and perhaps reprocess) the pipeline depending on what flags are set. That might be a handy way to reprocess a pipeline, but it does deviate from the conventions I've seen elsewhere (`kubectl` and `kops`).

What do you think?

Reply:
Good point, fixed.